### PR TITLE
Add support for the language Agda

### PR DESCRIPTION
**Pull request by xy-ren**
This PR adds support for the language [Agda](https://github.com/agda/agda).

### CODE REVIEW

File: components/prism-agda.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    'class-name': {
+      pattern: /((?:(?:data|record)) )[^\s{]+/,
+      lookbehind: true,
+    },
+    'function': {
+      pattern: /(\n\s*|^\s*)[^:\r\n]+?(?=:)/,
+      lookbehind: true,
+    },
+    'operator': {
+      pattern: /(^[(){}⦃⦄.;@\s]*|[(){}⦃⦄.;@\s]+)(?:[=|:∀→λ\\\?\_]|->)(?=[(){}⦃⦄.;@\s])/,
```
Comment (RunDevelopment) [inline components/prism-agda.js:22]: All punctuation tokens (`[(){}⦃⦄.;@]`) have already been matched, so we can simplify this pattern.
```suggestion
      pattern: /(^\s*|\s)(?:[=|:∀→λ\\?_]|->)(?=\s)/,
```

File: components/prism-agda.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+(function (Prism) {
+
+	Prism.languages.agda = {
+		'comment': [
+      /\{\-[\s\S]*?(?:\-\}|$)/,
+      /\-\-.*/,
+    ],
```
[the remark this answers is not in the document]
Comment (xy-ren) [reply components/prism-agda.js:7]: Thanks! I'll get this fixed.

File: components/prism-agda.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    'string': {
+      pattern: /"(?:\\(?:\r\n|[\s\S])|(?!")[^\\\r\n])*"/,
```
[the remark this answers is not in the document]
Comment (xy-ren) [reply components/prism-agda.js:9]: Thx!

File: components/prism-agda.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+      greedy: true,
+    },
+    'punctuation': /[(){}⦃⦄.;@]/,
+    'class-name': {
+      pattern: /((?:(?:data|record)) )[^\s{]+/,
```
[the remark this answers is not in the document]
Comment (xy-ren) [reply components/prism-agda.js:14]: Thx! There can be more than one spaces after `(?:data|record)`. Getting this fixed.

File: components/prism-agda.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+      lookbehind: true,
+    },
+    'function': {
+      pattern: /(\n\s*|^\s*)[^:\r\n]+?(?=:)/,
```
[the remark this answers is not in the document]
Comment (xy-ren) [reply components/prism-agda.js:18]: Thanks! In Agda, mostly all identifiers can be regarded as functions, so it's unreasonable to highlight all the functions... Here I only highlight function declarations (i.e. those followed by a `:`), and there can be more than 1 function names in a declaration.

File: components/prism-agda.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+      lookbehind: true,
+    },
+    'operator': {
+      pattern: /(^[(){}⦃⦄.;@\s]*|[(){}⦃⦄.;@\s]+)(?:[=|:∀→λ\\\?\_]|->)(?=[(){}⦃⦄.;@\s])/,
+      lookbehind: true,
+    },
+    'keyword': {
+      pattern: /\b(?:let|in|module|where|abstract|private|public|postulate|rewrite|with|record|constructor|field|open|infix|infixl|infixr|using|import|renaming|hiding|data|primitive|forall|Set|variable|tactic|instance|eta-equality|forall|inductive|macro|mutual|no-eta-equality|overlap|pattern|quote|quoteContext|quoteGoal|quoteTerm|syntax|unquote|unquoteDecl|unquoteDef)\b/,
+    },
```
[the remark this answers is not in the document]
Comment (xy-ren) [reply components/prism-agda.js:27]: Thanks! And yes, `Set` should be capital.

File: components/prism-agda.js (hunk 1 above)
Comment (xy-ren) [reply components/prism-agda.js:22]: Thanks!

**Comment by RunDevelopment** [comment]
Thank you very much for contributing @xy-ren!

**Pull request merged**